### PR TITLE
Usepbscluster

### PR DESCRIPTION
**Pull request by jedwards4b**
Hi Matt, Wonder if you could take a look at this and help me debug.   The first question is - why is pbs.py not recognizing my logger?   The second is that it doesn't seem to be submitting jobs to the queue, any idea why?

### CODE REVIEW

File: utilities/cheyenne/launch-dask.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import argparse
+import subprocess
+import re
+import time
+import logging
+import  socket
+from dask.distributed import Client
+from pangeo import PBSCluster
+
+global logger
```
Comment (mrocklin) [inline utilities/cheyenne/launch-dask.py:13]: You may not need to annotate this as global.  It's already top-level scope.

File: utilities/cheyenne/launch-dask.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    proc = subprocess.Popen(cmd)
+    dask_scheduler.jlab_proc = proc
+
+
+def get_logger(log_level):
+    ch = logging.StreamHandler(sys.stdout)
+    formatter = logging.Formatter(' - '.join(
+        ["%(asctime)s", "%(name)s", "%(levelname)s", "%(message)s"]))
+    ch.setFormatter(formatter)
+    logger = logging.getLogger(__file__)
```
Comment (mrocklin) [inline utilities/cheyenne/launch-dask.py:70]: I typically use `__name__` here rather than `__file__` much as you did above.  Perhaps this is your problem?

**Comment by mrocklin** [comment]
> The second is that it doesn't seem to be submitting jobs to the queue, any idea why?

Nothing strikes me at first glance.  I'll admit that I haven't taken a very deep look at this though.

File: utilities/cheyenne/launch-dask.py (hunk 2 above)
Comment (jedwards4b) [reply utilities/cheyenne/launch-dask.py:70]: I was initializing logger twice, but I removed this one and it still doesn't work.

**Comment by jedwards4b** [comment]
I found the problem.  ___exit__ is being called which in turn calls qdel and cancels the job.
If I comment out the exit call, I get a worker job and I can connect to a jupyter notebook but I 
don't seem to get the dask dashboard.   It seems like I just want this script to run and set things up and then exit, is your idea that this would persist as long as the dask job is running?

**Comment by mrocklin** [comment]
The PBSCluster object was intended for interactive use in the notebook.  It may not be appropriate and was not intended for setting up long running clusters.

**Comment by jedwards4b** [comment]
I still think that our objectives are similar.   I don't need the session to last any longer than the notebook does.   I have a script that sets up a dask scheduler and worker(s) on cheyenne and then connects a notebook to that, PBSCluster has some functionality that could make that easier.   
Are you saying that PBSCluster would allow you to start dask scheduler and workers from the notebook session directly?   I don't see a lot of difference except order of operations, and perhaps the ability to add or delete workers.  Maybe it would help if you provided an example notebook?

**Comment by mrocklin** [comment]
I've answered questions on https://github.com/pangeo-data/pangeo/pull/56

I'm going to close this for now.  For future comments and debugging help I recommend opening a PR on the `pangeo-data` fork so that others are aware and can engage in conversation.

**Pull request closed without merging**